src/app/buy-credits/buy-credits.component.ts

```diff
@@ -0,0 +1,245 @@
+    if (amountsGroup != null) {
+      this.amountsGroup = amountsGroup;
+    }
+
+    const giftAidGroup: any = this.creditForm.get('giftAid');
+    if (giftAidGroup != null) {
+      this.giftAidGroup = giftAidGroup;
+    }
+  }
+
+  buyCredits(): void {
+    this.isLoading = true;
+    const idAndJWT = this.identityService.getIdAndJWT();
+    if (idAndJWT !== undefined) {
+      this.identityService.getFundingInstructions(idAndJWT?.id, idAndJWT.jwt).subscribe((response: FundingInstruction) => {
+        this.isLoading = false;
+        this.isPurchaseComplete = true;
+        this.accountNumber = response.bank_transfer.financial_addresses[0].sort_code.account_number;
+        this.sortCode = response.bank_transfer.financial_addresses[0].sort_code.sort_code;
+        this.accountHolderName = response.bank_transfer.financial_addresses[0].sort_code.account_holder_name;
+      });
+    }
+  }
+
+  giftAidChoiceSelected(e: MatRadioChange) {
+    if (e.value === "yes") {
+      this.isOptedIntoGiftAid = true;
+    }
+    else {
+      this.isOptedIntoGiftAid = false;
+    }
+  }
+
+  showLoginDialog() {
+    const loginDialog = this.dialog.open(DonationStartLoginDialogComponent);
+    loginDialog.afterClosed().subscribe((data?: {id: string, jwt: string}) => {
+      if (data && data.id) {
+        this.loadAuthedPersonInfo(data.id, data.jwt);
+      }
+    });
+  }
+
+  customTip(): boolean {
+    return this.amountsGroup.value.tipPercentage === 'Other';
+  }
+
+  onTipSelectorChanged(e: MatSelectChange) {
+    if (e.value === 'Other') {
+      this.creditForm.get('amounts')?.get('customTipAmount')?.addValidators(Validators.required);
+
+    }
+
+    else {
+      this.creditForm.get('amounts')?.get('customTipAmount')?.removeValidators(Validators.required);
+    }
+
+    this.creditForm.get('amounts')?.get('customTipAmount')?.updateValueAndValidity();
+  }
+
+  giftAidAmount() : number {
+    // Gift Aid on the tip only when buying credits!
+    return this.calculatedTipAmount() * 0.25;
+  }
+
+  get creditAmountField() {
+    if (!this.creditForm) {
+      return undefined;
+    }
+
+    return this.creditForm.controls.amounts.get('creditAmount');
+  }
+
+  get customTipAmountField() {
+    if (!this.creditForm) {
+      return undefined;
+    }
+
+    return this.creditForm.controls.amounts.get('customTipAmount');
+  }
+
+  /**
+  * @returns Amount without any £/$s
+  */
+  sanitiseCurrency(amount: string): number {
+    return Number((amount || '0').replace('£', '').replace('$', ''));
+  }
+
+  calculatedTipAmount() : number {
+    const unsanitisedCreditAmount = this.amountsGroup.value.creditAmount;
+
+    if (!unsanitisedCreditAmount) {
+      return 0;
+    }
+
+    if (this.customTip()) {
+      const unsanitisedCustomTipAmount = this.amountsGroup.value.customTipAmount;
+      if (unsanitisedCustomTipAmount) {
+        return this.sanitiseCurrency(unsanitisedCustomTipAmount)
+      }
+      else {
```

Comment:
I think we can make this more readable by losing this "else", and the next one. `return` handles it and allows for less nesting + cognitive load.